tests/test_cmdline_crawl_with_pipeline/__init__.py

```diff
@@ -8,11 +8,16 @@ def _execute(self, spname):
         args = (sys.executable, "-m", "scrapy.cmdline", "crawl", spname)
         cwd = Path(__file__).resolve().parent
         proc = Popen(args, stdout=PIPE, stderr=PIPE, cwd=cwd)
-        proc.communicate()
-        return proc.returncode
+        _, stderr = proc.communicate()
+        return proc.returncode, stderr
 
     def test_open_spider_normally_in_pipeline(self):
-        assert self._execute("normal") == 0
+        returncode, stderr = self._execute("normal")
+        assert returncode == 0
 
     def test_exception_at_open_spider_in_pipeline(self):
-        assert self._execute("exception") == 1
+        returncode, stderr = self._execute("exception")
+        assert (
+            returncode == 0
+        )  # An unhandled exception in a pipeline should not stop the crawl
+        assert b'RuntimeError("exception")' in stderr
```

Comment:
This change in expectations may be worth discussing, and if it stays, I should probably add it to the list of backward incompatibilities.

Do we want *unhandled* exceptions in `SpiderMiddleware.open_spider` to prevent the spider from running normally? Should that apply to any handler of the [`spider_opened`](https://docs.scrapy.org/en/latest/topics/signals.html#std-signal-spider_opened) signal?